csvqb/csvqb/models/cube/csvqb/components/attribute.py

```diff
@@ -38,6 +43,44 @@ def new_attribute_values(self) -> List[NewQbAttributeValue]:
         pass
 
 
+@dataclass
+class QbAttributeLiteral(QbAttribute, ABC):
+    data_type: str = field(repr=False)
+
+    def validate_data(self, data: PandasDataTypes) -> List[ValidationError]:
+        accepted_data_types = [
```

Comment:
This feels more like a validation of the model rather than the data associated with it. So it doesn't feel like it should be in the `validate_data` method.

There's a nice way to apply validation like this using pydantic validators. See [this example](https://pydantic-docs.helpmanual.io/usage/validators/#validate-always).

Reply:
I think I've done this now.